src/ILCompiler.ReadyToRun/src/Compiler/ReadyToRunCompilerContext.cs

```diff
@@ -58,10 +58,14 @@ protected override bool ComputeHasGCStaticBase(FieldDesc field)
             Debug.Assert(field.IsStatic);
 
             TypeDesc fieldType = field.FieldType;
-            if (fieldType.IsValueType)
-                return ((DefType)fieldType).ContainsGCPointers;
+            if (fieldType.IsValueType && !fieldType.IsPrimitive)
```

Comment:
Nit: I would just move the `!fieldType.IsPrimitive` to the `return` clause (instead of `true`).

It will save us from inspecting `IsGCPointer` on value types...

Reply:
Good point, fixed in 2nd commit.